### PR TITLE
Graphs displaying data dynamically in widgets

### PR DESCRIPTION
**Pull request by BenCampbellNelson**
Overhaul of graph components with more dynamic y axis and x axis. 

Ability to scroll across graph on mobile view to prevent squishing!

All common logic between graph components moved to chartUtils.js 

Create variables for ease if need to change style and colouring of site. Buttons can be subject to change as well

### CODE REVIEW

**Review by SamanthaCra — CHANGES_REQUESTED** [review]
We'll need to increase the requests or find a way to disable the graphs being requests of their own. Looking through all of the different graphs just tanks the request limit.

**Comment by SamanthaCra** [comment]
Can we add buttons to display the graphs?

File: fullstack/src/app/page.js [hunk 1]
```diff
@@ -16,15 +16,31 @@ export default function Home() {
...
+          name="Air Pressure"
+          dataKey="pressure"
+          GraphComponent={LineChartComponent}
+        />
+        <Widget
+          name="Wind"
+          dataKey="wind"
+          GraphComponent={LineChartComponent}
+        />
+        <Widget name="CO2" dataKey="co2" GraphComponent={BarChartComponent} />
```
[the remark this answers is not in the document]
Comment (BenCampbellNelson) [reply fullstack/src/app/page.js:36]: Weird I didn't notice that, they were hourly

[the remark this answers is not in the document]
Comment (SamanthaCra) [reply fullstack/src/app/page.js:36]: image
Thanks for the fix :)

File: fullstack/src/components/graphs/LineChartComponent.jsx
Comment (SamanthaCra) [inline fullstack/src/components/graphs/LineChartComponent.jsx:1]: Nice and responsive

**Review by SamanthaCra — APPROVED** [review]
Awesome changes, looks really good. Nice friendly UI, responsive, good commenting and well thought out.

**Pull request merged**